src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDiffieHellmanCng.cs

```diff
@@ -55,6 +56,31 @@ private void ImportKeyBlob(byte[] ecKeyBlob, string curveName, bool includePriva
                 ForceSetKeySize(_key.KeySize);
             }
 
+            private void ImportLimitedPrivateKeyBlob(in ECParameters ecParams)
+            {
+                Debug.Assert(ecParams.Q.X is null && ecParams.Q.Y is null);
+                using AsnWriter writer = EccKeyFormatHelper.WritePkcs8PrivateKey(ecParams);
+                int length = writer.GetEncodedLength();
+                byte[] pkcs8Buffer = CryptoPool.Rent(length);
+
+                try
+                {
+                    if (!writer.TryEncode(pkcs8Buffer, out int encodedWritten))
```

Comment:
Meta, for AsnWriter API purposes: I wonder if some sort of `SendData<TState>(TState state, SpanAction<byte, TState> processor)` would be useful to avoid this ceremony without over-exposing the Span.

Reply:
Well, it turns out I didn't need any of this code anyway 😅.

I am curious why not make `EncodeAsSpan` public?

Reply:
Oh, were you thinking something like this?

```csharp
public void SendData<TState>(TState state, SpanAction<T, byte> processor) {
  // Validate
  processor(state, _buffer.AsSpan(0, _offset));
}
```

Yeah that would be useful, I would think.